src/main/scala/de/m7w3/signal/store/model/Groups.scala

```diff
@@ -1,56 +1,87 @@
 package de.m7w3.signal.store.model
 import java.util
 
+import de.m7w3.signal.store.DBActionRunner
 import org.whispersystems.signalservice.api.messages.multidevice.DeviceGroup
 import slick.driver.H2Driver.api._
 
 import scala.collection.JavaConverters.collectionAsScalaIterableConverter
 
-class Groups(tag: Tag) extends Table[(Int, Array[Byte], Option[String], Option[Array[Byte]], Boolean)](tag, "GROUPS") {
-  def pk = column[Int]("pk", O.PrimaryKey, O.AutoInc)
+case class Group(pk: Int,
+                 id: Array[Byte],
+                 name: Option[String],
+                 avatar: Option[Array[Byte]],
+                 active: Boolean) {
+  def members = {
+    GroupMembers.groupMembers.filter(_.groupId === pk).result
+  }
+}
+
+case class GroupWithMembers(group: Group,
+                            members: Seq[GroupMember])
+
+class Groups(tag: Tag) extends Table[Group](tag, "GROUPS") {
+  def pk = column[Int]("pk", O.PrimaryKey)
   def id = column[Array[Byte]]("ID")
   def name = column[Option[String]]("NAME")
   def avatar = column[Option[Array[Byte]]]("AVATAR")
   def active = column[Boolean]("ACTIVE", O.Default(true))
 
-  override def *  = (pk, id, name, avatar, active)
+  override def *  = (pk, id, name, avatar, active) <> (Group.tupled, Group.unapply)
 }
 
 object Groups {
   val groups = TableQuery[Groups]
 
   def get(id: Array[Byte]) = {
-    val join = groups join GroupMembers.groupMembers on (_.pk === _.groupId)
-    join.filter(joined => joined._1.id === id).result
+    val join = groups.filter(_.id === id) join GroupMembers.groupMembers on (_.pk === _.groupId)
+    join.result
+  }
+
+  def activeGroupsByName(dbActionRunner: DBActionRunner): Seq[GroupWithMembers] = {
+    val join = groups.filter(_.active === true).sortBy(_.name.asc) join GroupMembers.groupMembers on (_.pk === _.groupId)
+    val groupsAndMembers = dbActionRunner.run(join.result)
+    val grouped = groupsAndMembers.groupBy {
+      case (group, member) => group.pk
+    }
+    grouped.map { case (group, groupsMembers) =>
+      GroupWithMembers(groupsMembers.head._1, groupsMembers.map(_._2))
```

Comment:
groupsMembers contains the tuples of (Group, GroupMember) if this would be empty it wouldn't show up in the grouped map in the first place (smartypants)